File: 14.cpp
bool cmp(string a, string b){
    return a.length() < b.length();
}

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0)   return "";
        sort(strs.begin(), strs.end(), cmp);
        int minLen = strs[0].length();
        int len = 0, j;
        for (int i=0; i<minLen; i++){
            char ch = strs[0][i];
            for (j=1; j<strs.size(); j++){
                if (strs[j][i] != ch){
                    return strs[0].substr(0,len);
                }
            }
            if (j==strs.size() && strs[j-1][i] == ch){
                len++;
            }
        }
        return strs[0].substr(0,len);
    }
};

File: go-solutions/14.go
package main

import "sort"

func longestCommonPrefix(strs []string) string {
	switch len(strs) {
	case 0:
		return ""

	case 1:
		return strs[0]

	default:
		sort.Slice(strs, func(i, j int) bool {
			return len(strs[i]) < len(strs[j])
		})

		res := strs[0]
		for i := 1; i < len(strs); i++ {
			res = getLognestCommonPrefix(res, strs[i])
			if res == "" {
				break
			}
		}

		return res
	}
}

func getLognestCommonPrefix(str1, str2 string) string {
	var i int
	for i = 0; i < len(str1) && i < len(str2); i++ {
		if str1[i] != str2[i] {
			break
		}
	}

	return str1[:i]
}
